app/assets/javascripts/host_edit.js

```diff
@@ -205,10 +205,12 @@ function hostgroup_changed(element) {
   if (host_id) {
     if (host_changed ){
       update_form(element,{data:"&host[id]="+host_id});
-    } else if (host_changed == undefined) { // hostgroup changes parent
-      update_form(element);
-    } else { // edit host
-      set_inherited_value(element);
+    } else {
```

Comment:
I removed the `else if` because (lines 214-215):
```ruby
update_puppetclasses(element);
reload_host_params();
```
needed to be used in both cases (the `else if` and the `else`)